core/src/test/java/kafka/server/share/SharePartitionManagerTest.java

```diff
@@ -167,6 +167,7 @@ public class SharePartitionManagerTest {
     public void setUp() {
         mockTimer = new SystemTimerReaper("sharePartitionManagerTestReaper",
             new SystemTimer("sharePartitionManagerTestTimer"));
+        kafka.utils.TestUtils.clearYammerMetrics();
```

Comment:
we call `brokerTopicStats.close()` in the `teardown` method. However, other tests still seem to be affected by the metrics. Does this indicate that `brokerTopicStats.close()` does not effectively clean up all metrics?

Reply:
Though ideally it should, but added a safe check in setup as well. I cannot ever reproduce the tests failure on local, had more than 1000 runs.